File: bindings/lua/Polycode/ProgramAttribute.lua
class "ProgramAttribute"


function ProgramAttribute:__getvar(name)
	if name == "size" then
		return Polycode.ProgramAttribute_get_size(self.__ptr)
	elseif name == "name" then
		return Polycode.ProgramAttribute_get_name(self.__ptr)
	elseif name == "arrayType" then
		return Polycode.ProgramAttribute_get_arrayType(self.__ptr)
	end
end

function ProgramAttribute:__setvar(name,value)
	if name == "size" then
		Polycode.ProgramAttribute_set_size(self.__ptr, value)
		return true
	elseif name == "name" then
		Polycode.ProgramAttribute_set_name(self.__ptr, value)
		return true
	elseif name == "arrayType" then
		Polycode.ProgramAttribute_set_arrayType(self.__ptr, value)
		return true
	end
	return false
end
function ProgramAttribute:ProgramAttribute(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.ProgramAttribute(unpack(arg))
	end
end

function ProgramAttribute:__delete()
	if self then Polycode.delete_ProgramAttribute(self.__ptr) end
end

File: bindings/javascript/Polycode/ProgramAttribute.js
function ProgramAttribute() {
	Object.defineProperties(this, {
		'size': { enumerable: true, configurable: true, get: ProgramAttribute.prototype.__get_size, set: ProgramAttribute.prototype.__set_size},
		'name': { enumerable: true, configurable: true, get: ProgramAttribute.prototype.__get_name, set: ProgramAttribute.prototype.__set_name}
	})
}
ProgramAttribute.prototype.__get_size = function() {
	return Polycode.ProgramAttribute__get_size(this.__ptr)
}

ProgramAttribute.prototype.__set_size = function(val) {
	Polycode.ProgramAttribute__set_size(this.__ptr, val)
}

ProgramAttribute.prototype.__get_name = function() {
	return Polycode.ProgramAttribute__get_name(this.__ptr)
}

ProgramAttribute.prototype.__set_name = function(val) {
	Polycode.ProgramAttribute__set_name(this.__ptr, val)
}

Duktape.fin(ProgramAttribute.prototype, function (x) {
	if (x === ProgramAttribute.prototype) {
		return;
	}
	Polycode.ProgramAttribute__delete(x.__ptr)
})
